src/app/study-filters/study-filters.component.ts

```diff
@@ -41,17 +41,16 @@ export class StudyFiltersComponent extends StatefulComponent implements OnInit,
       datasets.filter(d => !d.parents.length).map(d => {
         datasetTrees.push(new DatasetNode(d, datasets));
       });
-      this.allStudyNodes = this.getAllStudyNodesById(datasetTrees, this.dataset.id);
+      this.allStudyNodes = findNodeById(datasetTrees[0], this.dataset.id);
```

Comment:
You have re-introduced a bug here. This does not work if there are multiple datasets on the top level. Rather, you should loop over `datasetTrees`, calling `findNodeById` on each element until you find the first matching node.